### PR TITLE
update openidcontext getStoredValue to check against NPE

### LINKED ISSUE #23061: NPE from getStoredValue when requested value doesn't exist
I have a test that uses an app protected by @OpenIdAuthenticationMechanismDefinition

When I reach the app, I'm trying to log the content of openIdContext.  I tried to obtain a value for an object that didn't exist using getStoredValue("sub").  getStoredValue is throwing an NPE.

```10/21/22, 13:04:46:152 CDT] 00000037 SystemOut                                                    O req: io.openliberty.webcontainer60.srt.SRTServletRequest60@6d259423
[10/21/22, 13:04:46:152 CDT] 00000037 SystemOut                                                    O resp: io.openliberty.webcontainer60.srt.SRTServletResponse60@5b114c4
[10/21/22, 13:04:46:152 CDT] 00000037 SystemOut                                                    O storedValue: sub
[10/21/22, 13:04:46:195 CDT] 00000037 com.ibm.ws.webcontainer.util.ApplicationErrorUtils           E SRVE0777E: Exception thrown by application class 'java.util.Objects.requireNonNull:221'
java.lang.NullPointerException
	at java.base/java.util.Objects.requireNonNull(Objects.java:221)
	at java.base/java.util.Optional.<init>(Optional.java:107)
	at java.base/java.util.Optional.of(Optional.java:120)
	at io.openliberty.security.jakartasec.identitystore.OpenIdContextImpl.getStoredValue(OpenIdContextImpl.java:236)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.jboss.weld.bean.proxy.AbstractBeanInstance.invoke(AbstractBeanInstance.java:38)
	at org.jboss.weld.bean.proxy.ProxyMethodHandler.invoke(ProxyMethodHandler.java:106)
	at org.jboss.weld.generated.proxies.security.enterprise.identitystore.openid.OpenIdContext$844015539$Proxy$_$$_WeldClientProxy.getStoredValue(Unknown Source)
	at oidc.client.base.utils.OpenIdContextLogger.logStoredValue(OpenIdContextLogger.java:233)
	at oidc.client.base.utils.OpenIdContextLogger.logStoredValues(OpenIdContextLogger.java:219)
	at oidc.client.base.utils.OpenIdContextLogger.logContext(OpenIdContextLogger.java:95)```

[NPE.zip](https://github.com/OpenLiberty/open-liberty/files/9841501/NPE.zip)

### PR DESCRIPTION
**Pull request by arunavemulapalli**
fixes #23061

### CODE REVIEW

**Comment by arunavemulapalli** [comment]
#build

**Comment by arunavemulapalli** [comment]
#build

**Comment by arunavemulapalli** [comment]
#libby

**Review by teddyjtorres — APPROVED** [review]

File: dev/io.openliberty.security.jakartasec.3.0.internal/src/io/openliberty/security/jakartasec/identitystore/OpenIdContextImpl.java [hunk 1]
```diff
@@ -233,7 +233,7 @@ public <T> Optional<T> getStoredValue(HttpServletRequest request, HttpServletRes
             value = (T) storage.get(storageName);
         }
 
-        return Optional.of(value);
+        return value != null ? Optional.of(value) : Optional.empty();
```
Comment (jimmy1wu) [inline dev/io.openliberty.security.jakartasec.3.0.internal/src/io/openliberty/security/jakartasec/identitystore/OpenIdContextImpl.java:236]: shorthand for what you are doing:
```suggestion
        return Optional.ofNullable(value);
```

Comment (arunavemulapalli) [reply dev/io.openliberty.security.jakartasec.3.0.internal/src/io/openliberty/security/jakartasec/identitystore/OpenIdContextImpl.java:236]: thanks

**Pull request merged**